acceptance: deal with all retry errors

tbg opened the pull request:
I don't think there's an issue open for it right now, but we're not
catching all of the error-after-context-timeout errors properly,
which caused me a bunch of red on CI this morning. Hopefully this
catches them all. I took the error messages from my local docker
client; they seem to match what I've seen, but we may need to adjust
them as docker gets upgraded and wording changes.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6035)

<!-- Reviewable:end -->

tbg committed:
acceptance: remove spurious error log

This was probably useful when building the retry-fix, but it is now
causing various log outputs because we call ContainerKill opportunistically
during cleanup (and ignore the error).

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-:-KFF2bZ6fHC7EAZs82rH:1417088721)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[acceptance/cluster/docker.go, line 403 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3G-_mKOsd9rtChwJ:-KFF3G-_mKOsd9rtChwK:634618649) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L403)):*
time to put a comment on the method to explain this.

---

*[acceptance/cluster/docker.go, line 410 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF2y6ZySznm-6bziXc:-KFF2y6_tDXZh0ZeU7TI:-515816306) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L410)):*
i don't believe this is logging what you think it is logging.

---

*[acceptance/cluster/docker.go, line 427 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3T0zA7daLwY8r-U0:-KFF38ef5aArpVvBb1mT:-703604238) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L427)):*
why is it safe to ignore this error? if the first attempt times out but doesn't create the container, this will give you a false success.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[acceptance/cluster/docker.go, line 403 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3G-_mKOsd9rtChwJ:-KFF5Vr6RaY6JxilIXFx:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L403)):*
Done.

---

*[acceptance/cluster/docker.go, line 410 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF2y6ZySznm-6bziXc:-KFF5oyR_xyQc1F6EOZ2:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L410)):*
Done.

---

*[acceptance/cluster/docker.go, line 427 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3T0zA7daLwY8r-U0:-KFF4GG_bYplXTrOlzGw:2037765807) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L427)):*
It's not safe, but the alternative is also shitty. I could remove the container before calling this - does that address your concern?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/docker.go, line 427 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3T0zA7daLwY8r-U0:-KFF6qtuMKXMToeHxHJF:1959919876) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L427)):*
I guess?

---

*[acceptance/cluster/docker.go, line 410 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF6gm7YMcAZxX2qTmG:-KFF6gm7YMcAZxX2qTmH:1125945286) ([raw file](https://github.com/cockroachdb/cockroach/blob/2bb3c43115b29ce06af3f22337a6f877329ce7c5/acceptance/cluster/docker.go#L410)):*
why don't you use regexp.MustCompile here? you're doing that by hand here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/docker.go, line 427 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF3T0zA7daLwY8r-U0:-KFFEIW6Tuy198rJpgc9:1849356840) ([raw file](https://github.com/cockroachdb/cockroach/blob/11f10c9f355e7ce8ef8e0b6f1e0bd698325c5760/acceptance/cluster/docker.go#L427)):*
Done. It's... not pretty. PTAL.

---

*[acceptance/cluster/docker.go, line 410 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFF6gm7YMcAZxX2qTmG:-KFFACrkSfXamefzfG-l:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/2bb3c43115b29ce06af3f22337a6f877329ce7c5/acceptance/cluster/docker.go#L410)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/cluster/docker.go, line 411 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFFK7vzAKRQ6vjaqKAP:-KFFK7w-qL3Pxw7qWOqA:2138286421) ([raw file](https://github.com/cockroachdb/cockroach/blob/f07abc5b23c03db014e67f729d8fd8f9d126ed41/acceptance/cluster/docker.go#L411)):*
get rid of `matched`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-:-KFFKYIwAmK2YS_7FKQG:1417088721)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/cluster/docker.go, line 411 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035#-KFFK7vzAKRQ6vjaqKAP:-KFFL_0bNMA8oWbyeK94:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/f07abc5b23c03db014e67f729d8fd8f9d126ed41/acceptance/cluster/docker.go#L411)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: deal with all retry errors

I don't think there's an issue open for it right now, but we're not
catching all of the error-after-context-timeout errors properly,
which caused me a bunch of red on CI this morning. Hopefully this
catches them all. I took the error messages from my local docker
client; they seem to match what I've seen, but we may need to adjust
them as docker gets upgraded and wording changes.

tamird commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6035)*
<!-- Sent from Reviewable.io -->
